src/index.tsx

```diff
@@ -73,11 +72,38 @@ var createWebmarkFolder = (): void => {
     )
 };
 
-var getCurrentUrl = (): string => {
-    return 'https://current.url.com';
+// type checker for proper string type url
+let typeCheckString = (text: string | undefined): string => {
+    if (text == undefined) {
+        // TODO: this block needs proper type guard rather than returning below url.
+        return "https://example.com/error";
+    } else {
+        return text;
+    }
 };
 
-var saveToWebmarkFolder = (url: string): void => {
+let getCurrentUrlAndSave = () => {
+    chrome.tabs.query(
+        { active: true, currentWindow: true },
+        ([currentTab]) => {
+            const thisTabUrl: string|undefined = currentTab.url;
+            const thisTabTitle: string|undefined = currentTab.title;
+            saveToWebmarkFolder(thisTabUrl, thisTabTitle);
+        });
+}
+
```

Comment:
Let's not keep more than one empty lines for now.--unless two blocks of code needs a clear demarcation. We will reconsider our rule later if it doesn't work for us. You can sneak this edit into another commit/issue.